src/sentry/api/helpers/group_index/update.py

```diff
@@ -547,7 +547,11 @@ def process_group_resolution(
             current_release_version = get_current_release_version_of_group(group, follows_semver)
 
             if current_release_version:
-                resolution_params.update({"current_release_version": current_release_version})
+                # Mypy believes `current_release_version` is of type `Release | None`,
+                # but we know it's of type `str` (CharField)
+                resolution_params.update(
+                    {"current_release_version": current_release_version}  # type: ignore[dict-item]
+                )
```

Comment:
it's because it is infering the type of `resolution_params` -- the correct fix is to make `resolution_params` a `TypedDict` with `current_release_version: NotRequired[str]`